### DISCUSSION TITLE
GraphQL filtering on Taxonomy

### DISCUSSION THREAD

**Discussion by PieEatingNinjas** [OrchardCMS/OrchardCore#6873] [Q&A] [answered]
Hi all!
I'm using Orchard to build a headless CMS where content editors should be able to create content that is consumed by a mobile app. So far, we are very pleased with Orchard, although -as newbies- we need to look up a lot of stuff 😄 
Currently, we're stuck in a particular scenario, that we can't seem to find a solution for. That's why I'm reaching out to the community.

In our scenario, the content can be classified in 1 or more 'categories'. We are using Taxonomy for this. Now we should be able to query the content items where a given 'category' should be one of the parameters along with some 'text' that needs to be found in the fulltext field. 

We are retrieving our data from Orchard using GraphQL. We've already build a Lucene query for the full text search that we call via GraphQL to get our results. So far, everything is going well. But, we don't have any clue how we could possibly add this additional category filter.

Hopefuly some smart people of the community can steer us in the right direction so we can build this!

Thanks!!

**Comment by Piedone** [comment]
I'm not sure I'm entirely following. So you have the content items with connected Taxonomy terms set up, right? And you have a Lucene index for the term titles (let's call the text there as such) and can retrieve these items via GraphQL. What is the missing step? Retrieving content items based on the title of their corresponding Taxonomies?

**Reply by PieEatingNinjas** [reply]
Yes, indeed. I would like to retrieve content items on text AND filter on the title of their taxonomy as well.
Imagine I search the term 'accident' and say the category is 'Belgian news' (my categories are the Taxonomies). In that case I don't want to get back articles containing 'accident' in the category 'US news'. But if there is an article containing 'accident' that has both 'Belgian news' and 'US news', I expect to get it back as well. The thing is, I can manually query this on the 'middle tier' in code, but this is far from ideal.
We currently have a good solution using SQL queries and some custom SQL views and that does the trick. But maybe there is a more out-of-the-box way of doing this?

**Comment by deanmarcussen** [comment, marked as answer]
Taxonomies are included and indexed by lucene.

Can you not include the taxonomy related information as a further filter on your full text query

**Reply by PieEatingNinjas** [reply]
So you suggest to put the labels of my taxonomies in the fulltext property of my contentitem?
I also thought of this, but it felt a little hacky... Because this would mean that I would have to put some 'structure' in the full text search property like '<content>{{ * content properties I want to search on}}</content><categories>{{ * catagories * }} </categories>' because if I would have a category 'foo' on which I would like to search, I don't want to have a match if 'foo' is only in my title (but not as category). Makes sence? Or am I seeing thing totaly wrong?

**Reply by deanmarcussen** [reply]
No I'm not suggesting that you add it to the full text property

Taxonomies are included as a field in lucene https://github.com/OrchardCMS/OrchardCore/blob/dev/src/OrchardCore.Modules/OrchardCore.Taxonomies/Indexing/TaxonomyFieldIndexHandler.cs

I'm suggesting you include more filters in your lucene query.

If that doesn't work for you then a custom sql query is also a reasonable way to go.

**Reply by PieEatingNinjas** [reply]
Thanks for your feedback! We are currently staying with our SQL query, but will certainly look into the info you provided as well.
